### PR TITLE
ActiveStorage tests: change load_defaults from 6.0 to the current version

### PR DESCRIPTION
**Pull request by crawler**
### Summary

While working on #42535 I noticed that the active storage dummy app loads default settings for 6.0 so i changed this to be ins sync with the current Rails.version and some of the tests became failing.
~~In this pull request i changed load_defaults to 6.1 and fixed tests that failed.~~
load_defaults is now set with the Rails MAJOR and MINOR version constants to always match current defaults.

### Other Information

~~7.0 configuration is failing because of #42471 and I'm don't aware of the details of this change. The ffmpeg is failing with default arguments introduced there. Nothing else is wrong.~~
Thanking @pixeltrix, everything is fine now.

### CODE REVIEW

**Comment by pixeltrix** [comment]
> 7.0 configuration is failing because of #42471 and I'm don't aware of the details of this change. The ffmpeg is failing with default arguments introduced there. Nothing else is wrong.

What version of ffmpeg do you have installed? The new defaults require version 3.4+

**Comment by crawler** [comment]
4.4 but it's failing not only for me. Please see:
https://buildkite.com/rails/rails/builds/78440#b60bc27e-54e5-4053-9b8e-e4ba2fc1e3f4

**Comment by pixeltrix** [comment]
@crawler okay, you need to change the default setting for `video_preview_arguments` to this:

```
"-vf 'select=eq(n\\,0)+eq(key\\,1)+gt(scene\\,0.015),loop=loop=-1:size=2,trim=start_frame=1' -frames:v 1 -f image2"
```

We need to use `\\` to ensure the backslash ends up in the ffmpeg command but we also need to wrap the argument to the `-vf` option in `''` so that the shell doesn't interpret it 😬 

It's probably easiest if you change your PR to use `load_defaults 7.0` and update the setting where it's mentioned in these files:

```
guides/source/configuring.md
railties/lib/rails/application/configuration.rb
railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_0.rb.tt
railties/test/application/configuration_test.rb
```

Thanks for spotting this 👍🏻

**Comment by SkipKayhil** [comment]
> It's probably easiest if you change your PR to use `load_defaults 7.0` 

Would it make sense to change the value to `Rails::VERSION::STRING.to_f` so that it doesn't require updating in the future? The same change could be made in [Action Mailbox](https://github.com/rails/rails/blob/main/actionmailbox/test/dummy/config/application.rb#L10) and [Action Text](https://github.com/rails/rails/blob/main/actiontext/test/dummy/config/application.rb#L11) as well

**Comment by crawler** [comment]
@pixeltrix Thank you for responding so fast! Did exactly what you said and now everything is fine. Changed the PR details.

**Comment by crawler** [comment]
@SkipKayhil I was very surprised when found that Dummy is hardcoded to load 6.0 defaults. I kinda feel dirty when I perform back and forth type conversion, so at first, I did it with Rails.version[..2] but this will cause a problem in 2024  😂 . Now it made via [MAJOR, MINOR].join(
".")

**Comment by SkipKayhil** [comment]
> Now it made via [MAJOR, MINOR].join(".")

I totally agree, the type conversion definitely looks a little weird. I initially suggested `Rails::VERSION::STRING.to_f` because there are a few usages of it in `load_defaults` related code already:
```shell
❯ rg -F 'Rails::VERSION::STRING.to_f'
lib/rails/generators/rails/app/app_generator.rb
266:      defined?(@config_target_version) ? @config_target_version : Rails::VERSION::STRING.to_f

lib/rails/generators/rails/app/templates/config/application.rb.tt
32:    config.load_defaults Rails::VERSION::STRING.to_f

test/generators/app_generator_test.rb
221:    assert_file "#{app_root}/config/application.rb", /\s+config\.load_defaults #{Rails::VERSION::STRING.to_f}/
486:      File.write(config, content.gsub(/config\.load_defaults #{Rails::VERSION::STRING.to_f}/, "config.load_defaults 5.1"))
```

Maybe @pixeltrix will have some thoughts, since dynamically getting the major/minor release from the `VERSION` is getting used in an increasing number of places it may be worth adding a dedicated method?

**Comment by zzak** [comment]
I think it's better to use `load_defaults 7.0`, this way we are aware of breaking changes that affect these types of tests.

I also started #42286 a while back but need to fix the failing tests.

File: activestorage/app/models/active_storage/variant_with_record.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # ActiveStorage::VariantRecord. This is only used if `ActiveStorage.track_variants` is enabled.
 class ActiveStorage::VariantWithRecord
   attr_reader :blob, :variation
+  delegate :service, to: :blob
```
Comment (zzak) [inline activestorage/app/models/active_storage/variant_with_record.rb:7]: Is this change necessary? 🤔

Comment (crawler) [reply activestorage/app/models/active_storage/variant_with_record.rb:7]: Since the VariantWithRcord was introduced a year ago I was confused why the common behavior was not extracted in some kind of a base class or an interface. Talking about this case, the read_image test helper is expecting that the variant responds to the service method. Is there is any problem with that?

Comment (zzak) [reply activestorage/app/models/active_storage/variant_with_record.rb:7]: @crawler Good catch, if there's a way to fix our tests without changing behavior that would be best.

I know there has been discussions on how to deal with these generated models, but no movement yet afaik.

**Comment by pixeltrix** [comment]
@crawler going to close this in favour of @zzak's PR since that addresses all of the dummy applications and was opened before this one.

**Comment by crawler** [comment]
@pixeltrix is there any chance that @zzak's PR will be merged any soon?

**Comment by pixeltrix** [comment]
Yes, I've just merged it

**Comment by crawler** [comment]
Cool! Because my another PR #42535 that fixes ActiveStorage controllers when the strict_loading_by_default is on for the ActiveRecord depends on this tests.

**Comment by crawler** [comment]
Can you please look into it?

**Comment by pixeltrix** [comment]
> Can you please look into it?

Will do later today as I have work commitments for the next few hours
